File: uploadservice/src/main/java/net/gotev/uploadservice/http/HttpStack.java
package net.gotev.uploadservice.http;

import java.io.IOException;

/**
 * Defines the methods that has to be implemented by an HTTP stack.
 * @author gotev (Aleksandar Gotev)
 */
public interface HttpStack {

    /**
     * Creates a new connection for a given URL and HTTP Method.
     * @param method HTTP Method
     * @param url URL to which to connect to
     * @return new connection object
     * @throws IOException if an error occurs while creating the connection object
     */
    HttpConnection createNewConnection(String method, String url) throws IOException;
}

File: uploadservice/src/main/java/net/gotev/uploadservice/network/HttpStack.kt
package net.gotev.uploadservice.network

import java.io.IOException

interface HttpStack {
    /**
     * Creates a new connection for a given URL and HTTP Method.
     * @param uploadId ID of the upload which requested this connection
     * @param method HTTP Method
     * @param url URL to which to connect to
     * @return new connection object
     * @throws IOException if an error occurs while creating the connection object
     */
    @Throws(IOException::class)
    fun newRequest(uploadId: String, method: String, url: String): HttpRequest
}
